rest_api/pyproject.toml

```diff
@@ -0,0 +1,72 @@
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "rest-api"
+description = 'API server for Haystack (https://github.com/deepset-ai/haystack)'
+readme = "README.md"
+requires-python = ">=3.7"
+license = "Apache-2.0"
+keywords = []
+authors = [
+  { name = "deepset.ai", email = "malte.pietsch@deepset.ai" },
+]
+classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Intended Audience :: Science/Research",
+  "Topic :: Scientific/Engineering :: Artificial Intelligence",
+  "Operating System :: OS Independent",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3.7",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: Implementation :: CPython",
+  "Programming Language :: Python :: Implementation :: PyPy",
+]
+dependencies = [
+    "farm-haystack",
+    "fastapi<1",
+    "uvicorn<1",
+    "gunicorn<21",
+    "python-multipart<1",  # optional FastAPI dependency for form data
+]
+dynamic = ["version"]
+
+[project.urls]
+Documentation = "https://github.com/unknown/rest-api#readme"
+Issues = "https://github.com/unknown/rest-api/issues"
+Source = "https://github.com/unknown/rest-api"
+
+[tool.hatch.version]
+path = "rest_api/__about__.py"
```

Comment:
Wow, this is new! 

I've googled it quickly and found some references in SO, but no official docs. Is it a style choice or something actually recommended for packages? I like it, and if it's a standard I'd like to see it in Haystack as well someday in place of the `VERSION.txt` file :star_struck:

Reply:
I don't think it's standard but it looks like a good practice. I got it from the default template used by `hatch` when you bootstrap a new project, it can be definitely ported to Haystack at some point